deploy/lib/ml_rest.rb

```diff
@@ -122,37 +173,55 @@ def install_extensions(path)
     def install_transforms(path)
       if (File.exists?(path))
 
-
         data = get_files(path)
         size = data.length
 
         data.each_with_index do |d, i|
-
           @logger.debug "Deploying #{File.basename(d)}"
-
-          if (File.extname(d).include?("xq"))
-            file_type = 'xquery'
-            headers = {
-                'Content-Type' => 'application/xquery'
-            }
-          elsif (File.extname(d).include?("xsl"))
-            file_type = 'xslt'
-            headers = {
-                'Content-Type' => 'application/xslt+xml'
-            }
-          end
-
+          
           file = open(d, "rb")
           contents = file.read
-          transformName = File.basename(d).gsub(/(.xqy|.xquery|.xq|.xslt|.xsl)$/, '')
+          
+          file_ext = File.extname(d)[1..-1]
+          file_name = File.basename(d, ".*")
+          
+          is_sjs = (file_ext == "sjs")
+          is_xsl = file_ext.include?("xsl")
+          is_xqy = file_ext.include?("xq")
+          
+          headers = {
+            'Content-Type' => (is_sjs ? 'application/vnd.marklogic-javascript' : (is_xsl ? 'application/xslt+xml': 'application/xquery'))
+          }
+          
+          transformName = file_name
           params = []
-
-
-          # TODO: I'm assuming there's a way to consolidate the following if/elsif but I'm a Ruby newbie and I'm being conservative
-          if (file_type == 'xquery')
+          
+          if (is_sjs)
```

Comment:
This might need a second look. Didn't have a .sjs transform at hand, not sure it works properly..